api/util/util.py

```diff
@@ -3,14 +3,19 @@
 from threading import Thread
 from typing import Generator, Iterable, TypeVar
 
+from django.conf import settings
+
 T = TypeVar("T")
 
 
 def postpone(function):
     def decorator(*args, **kwargs):
-        t = Thread(target=function, args=args, kwargs=kwargs)
-        t.daemon = True
-        t.start()
+        if settings.ENABLE_POSTPONE_DECORATOR:
```

Comment:
This is somewhat ugly, and I'd rather remove the `postpone` decorator entirely to rely instead upon the `run_in_thread` method exposed by registered tasks. But, I'd still then want to be able to disable the threading in a test environment because it's all a bit difficult to test anyway (and it's how we sometimes end up with errors in the logs, but not in the tests).